### PR TITLE
[RISCV] Form VFWMUL.VF when fp_extend is scalar and then splatted.

### PR DESCRIPTION
**Pull request by topperc**
Previously we only supported the extend being in the vector domain after the splat.

### CODE REVIEW

**Review by lukel97 — APPROVED** [review]

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -14518,6 +14525,24 @@ struct NodeExtensionHelper {
...
+      SDValue Op = OrigOperand.getOperand(1);
+      if (Op.getOpcode() != ISD::FP_EXTEND)
+        break;
+
+      unsigned NarrowSize = VT.getScalarSizeInBits() / 2;
+      unsigned ScalarBits = Op.getOperand(0).getValueSizeInBits();
+      if (NarrowSize != ScalarBits)
+        break;
+
+      SupportsFPExt = true;
```
Comment (lukel97) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:14543]: Do we want to set `EnforceOneUse = false` if RISCVISD::VMV_V_X_VL does it

**Review by preames — APPROVED** [review]
LGTM

Just for the record, this is making an assumption that we have a vf variant for all widening vv instructions, and wf variant for all wv instructions supported by this transform.  This appears to hold as far as I can tell.

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp (hunk 1 above)
Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:14543]: I'm not sure. For the integer case we use MaskedValueIsZero and ComputeMaxSignifcant bits so we never directly bypass a scalar zext/sext. That's a bit different than this combine which does bypass a scalar fp_extend.

**Comment by topperc** [comment]
> LGTM
> 
> Just for the record, this is making an assumption that we have a vf variant for all widening vv instructions, and wf variant for all wv instructions supported by this transform. This appears to hold as far as I can tell.

I guess I should add an FADD test too.

**Pull request merged**